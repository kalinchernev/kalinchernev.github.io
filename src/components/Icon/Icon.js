// @flow strict
import React from 'react';
import styles from './Icon.module.scss';

type Props = {
  name: string,
  icon: {
    viewBox?: string,
    path?: string,
  },
};

const Icon = ({ name, icon }: Props) => (
  <svg className={styles['icon']} viewBox={icon.viewBox}>
    <title>{name}</title>
    <path d={icon.path} />
  </svg>
);

export default Icon;
