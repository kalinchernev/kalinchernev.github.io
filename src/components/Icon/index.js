// @flow strict
export { default } from './Icon';
