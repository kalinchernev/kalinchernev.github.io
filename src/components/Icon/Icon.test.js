// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import Icon from './Icon';

describe('Icon', () => {
  const props = {
    name: 'test',
    icon: {
      viewBox: '0 0 0 0',
      path: '',
    },
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Icon {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
