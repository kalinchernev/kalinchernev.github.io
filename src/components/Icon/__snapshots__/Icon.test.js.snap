// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<svg
  className="icon"
  viewBox="0 0 0 0"
>
  <title>
    test
  </title>
  <path
    d=""
  />
</svg>
`;
